private/bufpkg/bufplugin/bufpluginconfig/testdata/success/go/buf.plugin.yaml

```diff
@@ -18,3 +18,4 @@ registry:
   opts:
     - separate_package=true
 spdx_license_id: Apache-2.0
+license_url: https://github.com/grpc/grpc-go/blob/master/LICENSE
```

Comment:
I only added `license_url` to this file, but not [`npm/buf.plugin.yaml`](https://github.com/bufbuild/buf/blob/main/private/bufpkg/bufplugin/bufpluginconfig/testdata/success/npm/buf.plugin.yaml) because A. I couldn't find the url to the license (though I could just use any url) B. It's an optional field and missing this field is still valid.

Reply:
The vast majority of the projects will use a standard license (where we won't need a custom URL and knowing the SPDX license id will allow us to build a URL). In that case, we'd only need to specify a license URL if using a non-standard license.

Do we want to consider updating the spdx-go-data command to surface additional attributes from the JSON file: https://github.com/spdx/license-list-data/blob/36873db191f0ce8b6e54839738aeb628a4bb9da9/json/licenses.json#L3442-L3445

https://raw.githubusercontent.com/spdx/license-list-data/v3.18/json/licenses.json

Reply:
Making the url optional sounds good. 

3.18 has [more licenses](https://github.com/spdx/license-list-data/blob/36873db191f0ce8b6e54839738aeb628a4bb9da9/json/licenses.json#L6247), but it seems both [3.12](https://github.com/bufbuild/buf/blob/main/private/pkg/spdx/cmd/spdx-go-data/main.go#L40) and 3.18 have the same amount of attributes for each licenses, a license from [3.12](https://raw.githubusercontent.com/spdx/license-list-data/v3.12/json/licenses.json) also has `see_also`:
```
{
      "reference": "./ODbL-1.0.json",
      "isDeprecatedLicenseId": false,
      "detailsUrl": "./ODbL-1.0.html",
      "referenceNumber": 342,
      "name": "Open Data Commons Open Database License v1.0",
      "licenseId": "ODbL-1.0",
      "seeAlso": [
        "http://www.opendatacommons.org/licenses/odbl/1.0/",
        "https://opendatacommons.org/licenses/odbl/1-0/"
      ],
      "isOsiApproved": false
}
```
We can still upgrade to 3.18 just for more licenses.

Reply:
Will generating the JSON file be part of this task?

I think we can do something like:

if the user has only provided id, display title and url by looking up in JSON
if the user has provided both id and url, display title from lookup, and the provided url
if the user has provided only url: display `LICENSE` or the provided (or something better) as title, and the provided url